pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -561,7 +574,7 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
      */
     function _upscaleArray(uint256[] memory amounts, uint256[] memory scalingFactors) internal view {
         for (uint256 i = 0; i < _getTotalTokens(); ++i) {
-            amounts[i] = Math.mul(amounts[i], scalingFactors[i]);
+            amounts[i] = FixedPoint.mulDown(amounts[i], scalingFactors[i]);
```

Comment:
I would add a note here of why we round down in the meantime. Later we can do a bit of more analysis if we need upscaleArrayUp/Down.

Reply:
I left a note in `_upscale`, since this function is the same except on an array.